test/functional/lib/tests.cpp

```diff
@@ -6756,20 +6750,19 @@ SetXdpRss(
     InitializeOidKey(&Key, OID_GEN_RECEIVE_SCALE_PARAMETERS, NdisRequestSetInformation);
     MpOidFilter(AdapterMp, &Key, 1);
 
-    TRY_RSS_SET_THREAD_CONTEXT Ctx;
+    struct TRY_RSS_SET_THREAD_CONTEXT {
+        HANDLE InterfaceHandle;
+        XDP_RSS_CONFIGURATION *RssConfig;
+        UINT32 RssConfigSize;
+    } Ctx;
     Ctx.InterfaceHandle = InterfaceHandle.get();
     Ctx.RssConfig = RssConfig.get();
     Ctx.RssConfigSize = RssConfigSize;
-    Ctx.Result = E_FAIL;
-    CXPLAT_THREAD_CONFIG ThreadConfig {
-        0, 0, NULL, TryRssSetFn, &Ctx
-    };
-    CXPLAT_THREAD AsyncThread;
-    TEST_TRUE(CXPLAT_SUCCEEDED(CxPlatThreadCreate(&ThreadConfig, &AsyncThread)));
-    auto DeleteThread = wil::scope_exit([&]
-    {
-        CxPlatThreadDelete(&AsyncThread);
-    });
+
+    CxPlatAsync Async([](void *Context) -> void* {
```

Comment:
One more thought: you could also template the return to eliminate all these `(void*)((uintptr_t)` casts.